Content.Shared/MobState/EntitySystems/SharedMobStateSystem.cs

```diff
@@ -272,20 +272,20 @@ protected virtual void ExitState(MobStateComponent component, DamageState? state
         /// <summary>
         ///     Updates the mob state..
         /// </summary>
-        public void UpdateState(MobStateComponent component, FixedPoint2 damage)
+        public void UpdateState(MobStateComponent component, FixedPoint2 damage, EntityUid? origin = null)
```

Comment:
should probably pass down the origin from the DamageChangedEvent listener (see line 178)